frame/pablo/src/lib.rs

```diff
@@ -981,6 +1012,79 @@ pub mod pallet {
 			lp_for_liquidity::<T>(pool, pool_account, base_amount, quote_amount)
 		}
 
+		fn update_accounts_deposited_one_asset_storage(
+			who: Self::AccountId,
+			pool_id: Self::PoolId,
+			lp_amount: Self::Balance,
+			action: SingleAssetAccountsStorageAction,
+		) -> Result<(), DispatchError> {
+			match action {
```

Comment:
Consider extracting out the actions into separate helper functions, this is quite a long function

Reply:
@benluelo Yes, you're right that it looks big, but it's because there are many cases which are needed to be covered. But the gist of function is very simple. And if separate it into different function, I suppose it'll be harder to read the code. What do you think?

Reply:
looking at it again this morning, I think there's quite a lot of room for simplification, so it should all be able to stay in one function without issue. (See the other thread)